### PR TITLE
[Fix] Improve the FID and eval hook

### PR DESCRIPTION
**Pull request by nbei**
This PR is to improve the implementation of FID and eval hook. The original implementation will cause redundant RAM cost, which may cause the weird stop by the system call without any error information.

### CODE REVIEW

File: mmgen/core/evaluation/eval_hooks.py [hunk 1]
```diff
@@ -108,7 +110,8 @@ def after_train_iter(self, runner):
         if rank == 0:
             sys.stdout.write('\n')
             for metric in self.metrics:
-                metric.summary()
+                with torch.no_grad():
```
Comment (plyfager) [inline mmgen/core/evaluation/eval_hooks.py:113]: I remember `@torch.no_grad()` has already been added to `summary` function.

Comment (nbei) [reply mmgen/core/evaluation/eval_hooks.py:113]: I just want to make sure the gradient feature map will not be kept. (In case of that users forget to add no_grad() ahead of summary)

**Review by plyfager — APPROVED** [review]
